### PR TITLE
Chapter 11 Account Activation

### PR DESCRIPTION
**Pull request by Iskandar19**

### CODE REVIEW

**Comment by Iskandar19** [comment]
Ready

File: app/assets/stylesheets/account_activations.scss [hunk 1]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the AccountActivations controller here.
```
Comment (ghost) [inline app/assets/stylesheets/account_activations.scss]: Delete this file please

File: app/helpers/account_activations_helper.rb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+module AccountActivationsHelper
```
Comment (ghost) [inline app/helpers/account_activations_helper.rb]: Delete this file please

File: test/controllers/account_activations_controller_test.rb [hunk 3]
```diff
@@ -0,0 +1,7 @@
+require "test_helper"
```
Comment (ghost) [inline test/controllers/account_activations_controller_test.rb]: Delete this file please

File: test/mailers/previews/user_mailer_preview.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
+# Preview all emails at http://localhost:3000/rails/mailers/user_mailer
+class UserMailerPreview < ActionMailer::Preview
```
Comment (ghost) [inline test/mailers/previews/user_mailer_preview.rb]: Delete this file please

File: test/mailers/user_mailer_test.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+require "test_helper"
```
Comment (ghost) [inline test/mailers/user_mailer_test.rb]: Delete this file please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by Iskandar19** [comment]
ready

File: config/environments/development.rb [hunk 6]
```diff
@@ -4,6 +4,19 @@
   # In the development environment your application's code is reloaded on
   # every request. This slows down response time but is perfect for development
   # since you don't have to restart the web server when you make code changes.
+  config.action_mailer.raise_delivery_errors = true
+  config.action_mailer.delivery_method = :smtp
+  config.action_mailer.default_url_options = {host: "localhost:3000" }
```
Comment (ghost) [inline config/environments/development.rb]:
```suggestion
  config.action_mailer.default_url_options = {host: "localhost:3000"}
```

File: app/models/user.rb [hunk 7]
```diff
@@ -1,5 +1,7 @@
 class User < ApplicationRecord
-  attr_accessor :remember_token
+  attr_accessor :remember_token, :activation_token
+  before_save :downcase_email
+  before_create :create_activation_digest
 
   before_save {self.email = email.downcase}
```
Comment (ghost) [inline app/models/user.rb]: please remove this line

File: app/models/user.rb [hunk 8]
```diff
@@ -31,9 +33,37 @@ def remember
...
+    UserMailer.account_activation(self).deliver_now
+  end
+
+  private
+
+  def downcase_email
+    self.email.downcase!
+  end
+
+  def create_activation_digest
```
Comment (ghost) [inline app/models/user.rb]: move this method above private please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by Iskandar19** [comment]
ready

**Comment by ngocvt-0484** [comment]
merged

**Pull request merged**